transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/ControlIT.java

```diff
@@ -0,0 +1,302 @@
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+/**
+ * RFC-6455, section 5.5 "Control Frames"
+ */
+public class ControlIT {
+
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/ws/control");
+
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/echo"))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("http://localhost:8001")
+                            .done()
+                        .done()
+                    .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+    
+    private final TestRule timeout = new DisableOnDebug(new Timeout(10, SECONDS));
+
+    @Rule
+    public final TestRule chain = outerRule(k3po).around(gateway).around(timeout);
+
+    @Test
+    @Specification({
+        "client.send.close.payload.length.0/handshake.request.and.frame"
+        })
+    public void shouldEchoClientCloseFrameWithEmptyPayload() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.close.payload.length.1/handshake.request.and.frame"
+        })
+    public void shouldEchoClientCloseFrameWithPayloadSize1() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.close.payload.length.125/handshake.request.and.frame"
+        })
+    public void shouldEchoClientCloseFrameWithPayload() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.close.payload.length.126/handshake.request.and.frame"
+        })
+    public void shouldFailWebSocketConnectionWhenClientSendCloseFrameWithPayloadTooLong() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.ping.payload.length.0/handshake.request.and.frame"
+        })
+    public void shouldPongClientPingFrameWithEmptyPayload() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.ping.payload.length.125/handshake.request.and.frame"
+        })
+    public void shouldPongClientPingFrameWithPayload() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.ping.payload.length.126/handshake.request.and.frame"
+        })
+    public void shouldFailWebSocketConnectionWhenClientSendPingFrameWithPayloadTooLong() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Ignore("Timeout error")
```

Comment:
Seems like a spec test issue. I wonder why it is expecting gateway to close the connection.